presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoStatement.java

```diff
@@ -248,6 +249,10 @@ final boolean internalExecute(String sql)
             currentWarningsManager.set(Optional.of(warningsManager));
             resultSet = new PrestoResultSet(client, maxRows.get(), progressConsumer, warningsManager);
 
+            for (Map.Entry<String, SelectedRole> entry : client.getSetRoles().entrySet()) {
```

Comment:
Also this file.

Reply:
https://github.com/prestodb/presto/pull/12383#issuecomment-467633410